pkg/asset/agent/image/interactive_disconnected_ignition.go

```diff
@@ -0,0 +1,212 @@
+package image
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	igntypes "github.com/coreos/ignition/v2/config/v3_2/types"
+	"github.com/coreos/stream-metadata-go/arch"
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/agent/common"
+	"github.com/openshift/installer/pkg/asset/agent/interactive"
+	"github.com/openshift/installer/pkg/asset/agent/manifests"
+	"github.com/openshift/installer/pkg/asset/agent/workflow"
+	"github.com/openshift/installer/pkg/asset/ignition"
+	"github.com/openshift/installer/pkg/asset/ignition/bootstrap"
+	"github.com/openshift/installer/pkg/version"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	interactiveDisconnectedIgnitionFilename = "interactive-disconnected-agent.ign"
+)
+
+// InteractiveDisconnectedIgnition is the asset responsible for generating the
+// ignition file required to support the interactive disconnected setup.
+type InteractiveDisconnectedIgnition struct {
+	Config *igntypes.Config
+	File   *asset.File
+}
+
+// Name returns the human-friendly name of the asset.
+func (i *InteractiveDisconnectedIgnition) Name() string {
+	return "Agent Installer Interactive Disconnected Ignition"
+}
+
+// Dependencies returns the assets on which the current asset depends.
+func (i *InteractiveDisconnectedIgnition) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&workflow.AgentWorkflow{},
+		&interactive.InstallConfig{},
+		&manifests.ClusterImageSet{},
+		&manifests.AgentPullSecret{},
+		&manifests.InfraEnv{},
+		&common.InfraEnvID{},
+	}
+}
+
+// Generate produces an asset used only for the interactive disconnected workflow.
+// In the connected UI, the only mandatory installation detail required is the release version (and architecture).
+// Optionally the user may specify the pull secret, an SSH key or the list of the OLM operators to be installed.
+// All the remaining configuration will be provided by the user subsequently in the disconnected environment,
+// through the assisted UI running on the rendezvous node.
+func (i *InteractiveDisconnectedIgnition) Generate(_ context.Context, dependencies asset.Parents) error {
```

Comment:
This looks very similar to the UnconfiguredIgnition.Generate. As Zane and Andrea have suggested, there could be a common builder or base ignition. 

One difference would be the unconfigured ignition would turn on agent-check-config-image.service and the interactive-disconnected would turn on the assisted-installer ui.